lib/ansible/module_utils/common/timeout.py

```diff
@@ -0,0 +1,50 @@
+# Copyright (c) 2020, Matt Martz <matt@sivel.net>
+# Simplified BSD License (see licenses/simplified_bsd.txt or https://opensource.org/licenses/BSD-2-Clause)
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+import signal
+from functools import wraps
+
+from ansible.module_utils.common.warnings import warn
+
+
+def _raise_timeout(signum, frame):
+    raise TimeoutError
+
+
+class Timeout:
```

Comment:
The docstring should probably explain that this is to be used in the main thread only... And the implementation should probably do the corresponding runtime checks.